src/proxy/ListProxy.ts

```diff
@@ -44,119 +44,198 @@ import {PartitionSpecificProxy} from './PartitionSpecificProxy';
 import {ClientMessage} from '../protocol/ClientMessage';
 import {
     ReadOnlyLazyList,
+    SchemaNotReplicatedError,
     UUID
 } from '../core';
 
 /** @internal */
 export class ListProxy<E> extends PartitionSpecificProxy implements IList<E> {
 
     add(element: E): Promise<boolean> {
-        return this.encodeInvoke(ListAddCodec, this.toData(element))
-            .then(ListAddAllCodec.decodeResponse);
+        try {
+            return this.encodeInvoke(ListAddCodec, ListAddAllCodec.decodeResponse, this.toData(element));
+        } catch (e) {
+            if (e instanceof SchemaNotReplicatedError) {
+                return this.registerSchema(e.schema, e.clazz).then(() => this.add(element));
+            }
+            return Promise.reject(e);
```

Comment:
> also, what do you think about putting try catch only around toData calls?

I tried that to places that can easily be done. However, did not bother with changing one liners. Should I do them? It will take some time, that's it. 

> there is a slight behavior change here. if the user were receiving a serialization error before, he will now receive a rejected promise from now on. i would say this is a good change smile but raising this to make sure that we won't miss it

Ok let me try to keep the behavior as it is.